backend/tracim_backend/lib/utils/request.py

```diff
@@ -192,215 +260,130 @@ def _cleanup(self, request: 'TracimRequest') -> None:
+        exception_if_invalid_id: Exception
+    ) -> int:
         """
-        Get current content from request
-        :param user: User who want to check the workspace
-        :param workspace: Workspace of the content
-        :param content: comment is related to this content
-        :param request: pyramid request
-        :return: current content
+        Get id from pyramid path
+        :param name: name of the parameter
+        :param exception_if_none: exception if no param found
+        :param exception_if_invalid_id: exception if id is not a correct integer
+        :return: id of parameter
         """
-        comment_id = ''
-        try:
-            if 'comment_id' in request.matchdict:
-                comment_id_str = request.matchdict['content_id']
-                if not isinstance(comment_id_str, str) or not comment_id_str.isdecimal():  # nopep8
-                    raise InvalidCommentId('comment_id is not a correct integer')  # nopep8
-                comment_id = int(request.matchdict['comment_id'])
-            if not comment_id:
-                raise ContentNotFoundInTracimRequest('No comment_id property found in request')  # nopep8
-            api = ContentApi(
-                current_user=user,
-                session=request.dbsession,
-                show_deleted=True,
-                show_archived=True,
-                config=request.registry.settings['CFG']
-            )
-            comment = api.get_one(
-                comment_id,
-                content_type=content_type_list.Comment.slug,
-                workspace=workspace,
-                parent=content,
-            )
-        except NoResultFound as exc:
-            raise ContentNotFound(
-                'Comment {} does not exist '
-                'or is not visible for this user'.format(comment_id)
-            ) from exc
-        return comment
-
-    def _get_current_content(
+        if name not in self.matchdict:
+            raise exception_if_none
+        id_param_as_str = self.matchdict[name]
+        if not isinstance(id_param_as_str, str) \
+                or not id_param_as_str.isdecimal():
+            raise exception_if_invalid_id
+        return int(id_param_as_str)
+
+    def _get_body_id(
             self,
-            user: User,
-            workspace: Workspace,
-            request: 'TracimRequest'
-    ) -> Content:
+            name: str,
+            exception_if_none: Exception,
+            exception_if_invalid_id: Exception
+    ) -> int:
         """
-        Get current content from request
-        :param user: User who want to check the workspace
-        :param workspace: Workspace of the content
-        :param request: pyramid request
-        :return: current content
+        Get id from pyramid json_body
+        :param name: name of the parameter
+        :param exception_if_none: exception if no param found
+        :param exception_if_invalid_id: exception if id is not a correct integer
```

Comment:
Why choose this design (exceptions given as parameter) ? If it is for technical reason, can you explain it in docstring ?

Reply:
As exception are not the same for each type (to be explicit, each one are handled differently by hapic) but mecanism to get id is the same, that the way in found to simplify code. Did you have any better idea ?

Reply:
No better idea ^^ That's ok thx.